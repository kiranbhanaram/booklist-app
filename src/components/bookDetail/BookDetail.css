.detail {
    background-color: #BD2A4E;
    width: 100%;
    height: 100%;
    padding: 40px 0;
    position: fixed;
    top: 0;
    color: white;
    left: 0;
    overflow: auto;
  }
  .detail.open {
    -moz-transform: translateX(0);
    -ms-transform: translateX(0);
    -webkit-transform: translateX(0);
    transform: translateX(0);
  }
   
  .detail-container {
    text-align: center;
    margin: 0 auto;
    padding: 40px;
    max-width: 500px;
  }
   
  .detail-container dl {
    margin: 0;
    padding: 0;
  }
   
  .detail-container dt {
    font-size: 2.2rem;
    font-weight: 300;
  }
   
  .detail-container dd {
    margin: 0 0 40px 0;
    font-size: 1.8rem;
    padding-bottom: 5px;
    border-bottom: 1px solid #ac2647;
    box-shadow: 0 1px 0 #c52c51;
  }
   
  .close {
    background: none;
    padding: 18px;
    color: #fff;
    font-weight: 300;
    border: 1px solid rgba(255, 255, 255, 0.4);
    border-radius: 4px;
    line-height: 1;
    font-size: 1.1rem;
    position: fixed;
    cursor: pointer;
    right: 40px;
    top: 20px;
    -moz-transition: border 0.3s linear;
    -o-transition: border 0.3s linear;
    -webkit-transition: border 0.3s linear;
    transition: border 0.3s linear;
  }
  .close:hover, .close:focus {
    background-color: #a82545;
    border: 1px solid #a82545;
  }