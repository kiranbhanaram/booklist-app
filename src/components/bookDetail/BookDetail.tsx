import './BookDetail.css';
import { BookProps } from '../booklist/BookList';

type LocationParams = {
  rowData: BookProps;
  onClose: () => void;
};

export function BookDetail(props: LocationParams) {
  const { language, direction, source, comments, name } = props.rowData;
  return (
    <div className="detail">
      <div className="detail-container">
        <div>
          <label>Name</label>
          <div id="detail-name">{name || '-'}</div>
        </div>

        <div>
          <label>Language</label>
          <div id="detail-lang">{language || '-'}</div>
        </div>

        <div>
          <label>Direction</label>
          <div id="detail-direction">{direction || '-'}</div>
        </div>

        <div>
          <label>Source</label>
          <div id="detail-source">{source || '-'}</div>
        </div>

        <div>
          <label>Comments</label>
          <div id="detail-comment">{comments || '-'}</div>
        </div>
      </div>
      <div className="detail-nav">
        <button className="close" onClick={() => props.onClose()}>
          Close
        </button>
      </div>
    </div>
  );
}
