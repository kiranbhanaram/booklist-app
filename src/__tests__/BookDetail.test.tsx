import { shallow } from 'enzyme';
import { BookDetail } from '../components/bookDetail/BookDetail';

it('renders BookDetail component', () => {
  expect(
    shallow(
      <BookDetail
        rowData={{
          name: undefined,
          author: undefined,
          language: '',
          direction: '',
          source: '',
          comments: '',
        }}
        onClose={function (): void {
          throw new Error('Function not implemented.');
        }}
      />
    )
  ).toMatchSnapshot();
});

it('Renders props data of Book Detail component', () => {
  const data = {
    name: 'ace-tgkhmahjiddinju',
    language: 'Achinese',
    direction: 'ltr',
    source: 'https://www.islamic-invitation.com/',
    comments:
      'This translation was OCRed, in case you find any mistakes, let me know at https://bit.ly/33FuNZS',
  };

  const onClose = (): void => {
    throw new Error('Function not implemented.');
  };

  // Shallow render Book Detail component
  const wrapper = shallow(<BookDetail rowData={data} onClose={onClose} />);

  expect(wrapper.find('#detail-name').text()).toEqual(data.name);
  expect(wrapper.find('#detail-lang').text()).toEqual(data.language);
  expect(wrapper.find('#detail-direction').text()).toEqual(data.direction);
  expect(wrapper.find('#detail-source').text()).toEqual(data.source);
  expect(wrapper.find('#detail-comment').text()).toEqual(data.comments);

  expect(wrapper.find('.close').length).toEqual(1);
  wrapper.find('.close').simulate('onClick');
});
