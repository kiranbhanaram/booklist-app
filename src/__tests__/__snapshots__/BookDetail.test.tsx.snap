// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BookDetail component 1`] = `ShallowWrapper {}`;
